File: assembly-tests/fibo.c
/* Written by Luca Saiu in 2013
   I hereby release this file into the public domain, up to the extent
   permitted by the law. */

#include <stdio.h>

long fibo(long n) __attribute__((noinline));
long fibo(long n){
  if((n == 0) || (n == 1))
    return n;
  else
    return fibo(n - 2) + fibo(n - 1);
}


int main(void){
  printf("%i\n", (int)fibo(40));
  return 0;
}

File: assembly-tests/fibo.ml
(* Written by Luca Saiu in 2013
   I hereby release this file into the public domain, up to the extent
   permitted by the law. *)

let rec fibo n =
  if n < 2 then
    n
  else
    (fibo (n - 2)) + (fibo (n - 1))

let _ = Printf.printf "%i\n" (fibo 40)
